cluster/store.go

```diff
@@ -21,3 +32,31 @@ func NewKVStore(ctx context.Context, etcdAddr string) (*KVStore, error) {
 		kv: clientv3.NewKV(c),
 	}, nil
 }
+
+// Get returns the best matched value for the key provided
+func (kvs *KVStore) Get(ctx context.Context, key string) (string, error) {
+	getres, err := kvs.kv.Get(ctx, fmt.Sprintf("%s/%s", storePrefix, key), defaultGetOptions...)
+	if err != nil {
+		return "", fmt.Errorf("failed to get key %s: %w", key, err)
+	}
+
+	if len(getres.Kvs) == 0 {
+		return "", ErrNoKey
+	}
+	return string(getres.Kvs[0].Value), nil
+}
+
+// GetAll returns all values that correspond to the supplied key
+func (kvs *KVStore) GetAll(ctx context.Context, key string) ([]string, error) {
+	getres, err := kvs.kv.Get(ctx, fmt.Sprintf("%s/%s", storePrefix, key), defaultGetOptions...)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get key %s: %w", key, err)
+	}
+
+	gets := make([]string, 0)
+	for _, kvs := range getres.Kvs {
```

Comment:
I think it's a no key error if the list is empty right?

Reply:
I can add it but I don't think it will be useful, I guess it just makes the API more consistent.